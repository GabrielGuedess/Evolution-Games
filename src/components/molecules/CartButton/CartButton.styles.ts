import * as Popover from '@radix-ui/react-popover';

import styled, { css } from 'styled-components';

export const Wrapper = styled.div``;

export const Arrow = styled(Popover.Arrow)`
  ${({ theme }) => css`
    fill: ${theme.colors.gameDetails};
  `}
`;

export const Content = styled(Popover.Content)`
  width: 100%;
  margin: 0 1.6rem;
`;
