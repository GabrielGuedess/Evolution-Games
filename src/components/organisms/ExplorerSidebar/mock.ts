const filterCategories = {
  title: 'Genre',
  name: 'genre',
  type: 'checkbox',
  fields: [
    { title: 'Action-adventure', name: 'action-adventure' },
    { title: 'Adventure', name: 'adventure' },
    { title: 'Sports', name: 'sports' },
    { title: 'Puzzle', name: 'puzzle' },
    { title: 'Horror', name: 'horror' },
    { title: 'Platform', name: 'platform' },
    { title: 'Fantasy', name: 'fantasy' },
    { title: 'RPG', name: 'role-playing' },
    { title: 'JRPG', name: 'jrpg' },
    { title: 'Simulation', name: 'simulation' },
    { title: 'Strategy', name: 'strategy' },
    { title: 'Shooter', name: 'shooter' },
  ],
};

const filterYear = {
  title: 'Year',
  name: 'year',
  type: 'checkbox',
  fields: [
    { title: '1990 - 2000', name: '1990-2000' },
    { title: '2001 - 2005', name: '2001-2005' },
    { title: '2006 - 2011', name: '2006-2011' },
    { title: '2012 - 2017', name: '2012-2017' },
    { title: '2018 - Now', name: '2018-now' },
  ],
};

const filterPlatforms = {
  title: 'Platforms',
  name: 'platforms',
  type: 'checkbox',
  fields: [
    {
      title: 'Playstation 4',
      name: 'ps4',
    },
    {
      title: 'Playstation 5',
      name: 'ps5',
    },
    {
      title: 'Xbox One',
      name: 'one',
    },
    {
      title: 'Xbox Series X',
      name: 'xs',
    },
  ],
};

const filterFeedback = {
  title: 'Notas',
  name: 'feedback',
  type: 'checkbox',
  fields: [
    { title: 'From 5.0', name: 'from-5.0' },
    { title: 'From 4.0', name: 'from-4.0' },
  ],
};

export const filterItems = [
  filterCategories,
  filterYear,
  filterPlatforms,
  filterFeedback,
];
