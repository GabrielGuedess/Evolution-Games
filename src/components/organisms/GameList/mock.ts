import { GameCardProps } from 'components/molecules/GameCard/GameCard';

const gameCardItems: GameCardProps[] = [
  {
    id: '1',
    title: 'God of War: Ragnar√∂k',
    genre: 'Action-adventure ',
    developer: 'Santa Monica Games',
    year: '2022',
    img: 'https://i.pinimg.com/originals/34/f8/e5/34f8e55845212171e8bb35edc5b08018.jpg',
    price: '$ 60.99',
    score: 5.0,
    platform: ['playstation'],
    slug: '',
    primaryColor: '#9db4d0',
  },
  {
    id: '2',
    title: 'Star Wars Jedi: Fallen Order',
    genre: 'Action-adventure ',
    developer: 'EA',
    year: '2019',
    img: 'https://cdn2.steamgriddb.com/file/sgdb-cdn/grid/935a281a70ae21d54479c9c468145f7e.png',
    price: '$ 30.00',
    score: 3.8,
    platform: ['playstation', 'pc', 'xbox'],
    slug: '',
    primaryColor: '#a8b6cb',
  },
  {
    id: '3',
    title: 'Cyberpunk 2077',
    genre: 'Action-adventure ',
    developer: 'CD PROJEKT RED',
    year: '2020',
    img: 'https://whatifgaming.com/wp-content/uploads/2022/05/Neon-Aesthetic.jpg',
    price: '$ 10.00',
    score: 1.8,
    platform: ['playstation', 'pc', 'xbox'],
    slug: '',
    primaryColor: '#dabbde',
  },
  {
    id: '4',
    title: 'The Last of Us Part II',
    genre: 'Action-adventure ',
    developer: 'Naughty Dog',
    year: '2020',
    img: 'https://www.jeuxvideo-live.com/wp-content/uploads/jvl/2022/07/le-multi-de-the-last-of-us-2-deevoile-quelques-secrets-de-conception-104207-large.jpg',
    price: '$ 60.99',
    score: 5.0,
    platform: ['playstation'],
    slug: '',
    primaryColor: '#b2a7a5',
  },
  {
    id: '5',
    title: 'Elden Ring',
    genre: 'Action-adventure ',
    developer: 'BANDAI NAMCO Entertainment Inc',
    year: '2022',
    img: 'https://image.api.playstation.com/vulcan/ap/rnd/202110/2000/aGhopp3MHppi7kooGE2Dtt8C.png',
    price: '$ 60.99',
    score: 4.8,
    platform: ['playstation'],
    slug: '',
    primaryColor: '#aeb195',
  },
  {
    id: '6',
    title: 'Horizon Forbidden West',
    genre: 'Action-adventure ',
    developer: 'Santa Monica Games',
    year: '2022',
    img: 'https://t.ctcdn.com.br/efyy2ADqd67kPvUfXBVJWMJiX_I=/1400x788/smart/filters:format(webp)/i499600.png',
    price: '$ 60.99',
    score: 4.8,
    platform: ['playstation'],
    slug: '',
    primaryColor: '#aeb1d2',
  },
  {
    id: '7',
    title: "Marvel's Spider-Man",
    genre: 'Action-adventure ',
    developer: 'Insomniac Games',
    year: '2018',
    img: 'https://image.api.playstation.com/vulcan/ap/rnd/202008/1020/b0OfuUs4lHGWG4VNlHcXwL5a.jpg',
    price: '$ 60.99',
    score: 4.7,
    platform: ['playstation', 'pc'],
    slug: '',
    primaryColor: '#9ab3c5',
  },
  {
    id: '8',
    title: 'The Witcher 3',
    genre: 'Action-adventure ',
    developer: 'CD PROJEKT SA',
    year: '2015',
    img: 'https://e.snmc.io/lk/f/x/1508cc32995d91b6206ee1b1e82b9338/8329790',
    price: '$ 60.99',
    score: 4.5,
    platform: ['playstation', 'pc', 'xbox'],
    slug: '',
    primaryColor: '#d1b2a3',
  },
];

export default gameCardItems;
