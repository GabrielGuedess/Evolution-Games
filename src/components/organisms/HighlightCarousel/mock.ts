import { HighlightGameProps } from 'components/molecules/HighlightGame/HighlightGame';

export const highlightMock: HighlightGameProps[] = [
  {
    img: '/img/games/starWarsHighlight.png',
    gameLogo: '/img/games/highlightName/nameStarWars.png',
    gameLogoAlt: 'Logo Star Wars',
    alt: 'Star Wars',
    releaseYear: new Date(2019, 10, 10),
    genre: 'Action-adventure',
    developer: 'Respawn Entertainment',
    platform: ['pc', 'playstation', 'xbox'],
  },
  {
    img: '/img/games/r8.jpg',
    gameLogo: '/img/games/highlightName/nameR8.png',
    gameLogoAlt: 'Logo Star Wars',
    alt: 'Resident Evil 8',
    releaseYear: new Date('09-10-2022'),
    genre: 'Action-adventure',
    developer: 'Respawn Entertainment',
    platform: ['pc', 'playstation', 'xbox'],
  },
  {
    img: '/img/games/starWars.jpg',
    gameLogo: '/img/games/highlightName/nameStarWars.png',
    gameLogoAlt: 'Logo Star Wars 2',
    alt: 'Star Wars 2',
    releaseYear: new Date('09-10-2022'),
    genre: 'Action-adventure',
    developer: 'Respawn Entertainment',
    platform: ['pc', 'playstation', 'xbox'],
  },
  {
    img: '/img/games/theLastOfUs.jpg',
    gameLogo: '/img/games/highlightName/nameTheLastOfUs.png',
    gameLogoAlt: 'Logo The last Of Us',
    alt: 'The last Of Us',
    releaseYear: new Date('09-10-2022'),
    genre: 'Action-adventure',
    developer: 'Respawn Entertainment',
    platform: ['playstation'],
  },
];
