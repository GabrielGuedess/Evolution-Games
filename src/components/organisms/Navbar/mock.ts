import { GameCartDropProps } from 'components/molecules/GameCartDrop/GameCartDrop';

export const CartItems: GameCartDropProps[] = [
  {
    src: 'https://images.unsplash.com/photo-1662414590066-91c548626702?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyNjIxMDF8MHwxfHJhbmRvbXx8fHx8fHx8fDE2NjMyOTQxNjQ&ixlib=rb-1.2.1&q=80&w=1080',
    genre: 'Action',
    title: 'The Witcher 3',
    developer: 'CD Projekt',
    price: 100,
  },
  {
    src: 'https://images.unsplash.com/photo-1660677727122-ba176d2484eb?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyNjIxMDF8MHwxfHJhbmRvbXx8fHx8fHx8fDE2NjMyOTgwMzk&ixlib=rb-1.2.1&q=80&w=1080',
    genre: 'Action',
    title: 'GTA 6',
    developer: 'CD Projekt',
    price: 100,
  },
  {
    src: 'https://images.unsplash.com/photo-1662747923631-cbb301f15c5f?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyNjIxMDF8MHwxfHJhbmRvbXx8fHx8fHx8fDE2NjMyOTgwNTc&ixlib=rb-1.2.1&q=80&w=1080',
    genre: 'Action',
    title: 'Rainbow Six',
    developer: 'CD Projekt',
    price: 100,
  },
];
