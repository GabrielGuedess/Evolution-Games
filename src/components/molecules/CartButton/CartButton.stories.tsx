import { Story, Meta } from '@storybook/react/types-6-0';

import { cartItems } from 'components/organisms/Navbar/mock';

import { CartDropDownProps } from '../CartDropDown/CartDropDown';
import { CartButton } from './CartButton';

export default {
  title: 'Molecules/CartButton',
  component: CartButton,
  args: {
    items: cartItems,
  },
} as Meta<CartDropDownProps>;

export const Default: Story<CartDropDownProps> = args => (
  <CartButton {...args} />
);
