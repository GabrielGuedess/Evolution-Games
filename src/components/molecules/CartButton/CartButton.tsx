import Link from 'next/link';

import { useCart } from 'hooks/useCart/useCart';

import { CartIcon } from 'components/atoms/CartIcon/CartIcon';

import * as S from './CartButton.styles';

export const CartButton = () => {
  const { quantity } = useCart();

  return (
    <Link href="/cart">
      <S.Wrapper>
        <CartIcon quantity={quantity} />
      </S.Wrapper>
    </Link>
  );
};
