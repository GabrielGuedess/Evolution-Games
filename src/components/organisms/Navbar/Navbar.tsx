import Link from 'next/link';
import { useRouter } from 'next/router';

import { useState } from 'react';

import * as NavigationMenu from '@radix-ui/react-navigation-menu';
import * as Popover from '@radix-ui/react-popover';
import { Squash as Hamburger } from 'hamburger-react';
import { Heart } from 'phosphor-react';

import { Logo } from 'components/atoms/Logo/Logo';
import MediaMatch from 'components/atoms/MediaMatch/MediaMatch';
import {
  AvatarUser,
  AvatarUserProps,
} from 'components/molecules/AvatarUser/AvatarUser';
import { CartButton } from 'components/molecules/CartButton/CartButton';
import { MenuMobile } from 'components/molecules/MenuMobile/MenuMobile';

import { CartItems } from './mock';
import * as S from './Navbar.styles';

export type MenuProps = {
  loading?: boolean;
} & AvatarUserProps;

export const Navbar = ({ loading = false, ...props }: MenuProps) => {
  const { asPath } = useRouter();
  const [isOpen, setIsOpen] = useState(false);

  return (
    <S.Wrapper>
      <Logo />

      <MediaMatch greaterThan="medium">
        <NavigationMenu.Root>
          <S.ListTextLink>
            <S.ItemTextLink>
              <Link href="/" passHref>
                <S.LinkTextLink active={asPath === '/'}>Home</S.LinkTextLink>
              </Link>
            </S.ItemTextLink>

            <S.ItemTextLink>
              <Link href="/games" passHref>
                <S.LinkTextLink active={asPath === '/games'}>
                  Explorer
                </S.LinkTextLink>
              </Link>
            </S.ItemTextLink>

            <S.ItemTextLink>
              <Link href="/games" passHref>
                <S.LinkTextLink active={asPath === '/games'}>
                  Contact
                </S.LinkTextLink>
              </Link>
            </S.ItemTextLink>

            <NavigationMenu.Indicator />
          </S.ListTextLink>

          <NavigationMenu.Viewport />
        </NavigationMenu.Root>
      </MediaMatch>

      {!loading && (
        <>
          <S.WrapperIcons>
            <Link href="/favorites" passHref>
              <S.HeartLink aria-label="Link Favorites">
                <Heart size={24} />
              </S.HeartLink>
            </Link>

            <CartButton items={CartItems} />

            <MediaMatch greaterThan="medium">
              <AvatarUser {...props} />
            </MediaMatch>
          </S.WrapperIcons>

          <MediaMatch lessThan="medium">
            <Popover.Root open={isOpen} onOpenChange={setIsOpen}>
              <Popover.Trigger>
                <Hamburger
                  size={23}
                  duration={0.5}
                  label="Show menu"
                  toggled={isOpen}
                  toggle={setIsOpen}
                />
              </Popover.Trigger>
              <Popover.Anchor />
              <MenuMobile path={asPath} {...props} />
            </Popover.Root>
          </MediaMatch>
        </>
      )}
    </S.Wrapper>
  );
};
