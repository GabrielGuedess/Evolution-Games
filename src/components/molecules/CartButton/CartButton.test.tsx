import { renderWithTheme } from 'utils/tests/helpers';

import { screen } from '@testing-library/react';

import { CartItems } from 'components/organisms/Navbar/mock';

import { CartButton } from './CartButton';

jest.mock('components/atoms/CartIcon/CartIcon', () => ({
  __esModule: true,
  CartIcon: function Mock({ children }: { children: React.ReactNode }) {
    return <div data-testid="Mock CartIcon">{children}</div>;
  },
}));

describe('<CartButton />', () => {
  it('should render others components correctly', () => {
    // Arrange
    const { container } = renderWithTheme(<CartButton items={CartItems} />);

    // Assert
    expect(screen.getByTestId('Mock CartIcon')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
