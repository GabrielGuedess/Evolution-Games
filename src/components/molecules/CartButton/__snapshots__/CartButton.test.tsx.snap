// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<CartButton /> should render others components correctly 1'] = `
<div
  class=""
>
  <button
    aria-controls="radix-:r0:"
    aria-expanded="false"
    aria-haspopup="dialog"
    data-state="closed"
    type="button"
  >
    <div
      data-testid="Mock CartIcon"
    />
  </button>
  <div />
</div>
`;
